src/test/regress/citus_tests/common.py

```diff
@@ -315,8 +347,711 @@ def eprint(*args, **kwargs):
+            self.sql(
+                "SELECT pg_drop_replication_slot(slot_name) FROM pg_replication_slots WHERE slot_name = %s",
+                (slot,),
+            )
+
+    def cleanup_subscriptions(self):
+        for subscription in self.subscriptions:
+            try:
+                self.sql(
+                    sql.SQL("ALTER SUBSCRIPTION {} DISABLE").format(
+                        sql.Identifier(subscription)
+                    )
+                )
+            except psycopg.errors.UndefinedObject:
+                # Subscription didn't exist already
+                continue
+            self.sql(
+                sql.SQL("ALTER SUBSCRIPTION {} SET (slot_name = NONE)").format(
+                    sql.Identifier(subscription)
+                )
+            )
+            self.sql(
+                sql.SQL("DROP SUBSCRIPTION {}").format(sql.Identifier(subscription))
+            )
+
+    @contextmanager
+    def transaction(self, **kwargs):
+        with self.cur(**kwargs) as cur:
+            with cur.connection.transaction():
+                yield cur
+
+    def sleep(self, duration=3, **kwargs):
+        """Run pg_sleep"""
+        return self.sql(f"select pg_sleep({duration})", **kwargs)
+
+    def asleep(self, duration=3, times=1, sequentially=False, **kwargs):
+        """Run pg_sleep asynchronously in a task.
+
+        times:
+            You can create a single task that opens multiple connections, which
+            run pg_sleep concurrently. The asynchronous task will only complete
+            once all these pg_sleep calls are finished.
+        sequentially:
+            Instead of running all pg_sleep calls spawned by providing
+            times > 1 concurrently, this will run them sequentially.
+        """
+        return asyncio.ensure_future(
+            self.asleep_coroutine(
+                duration=duration, times=times, sequentially=sequentially, **kwargs
+            )
+        )
+
+    async def asleep_coroutine(self, duration=3, times=1, sequentially=False, **kwargs):
+        """This is the coroutine that the asleep task runs internally"""
+        if not sequentially:
+            await asyncio.gather(
+                *[
+                    self.asql(f"select pg_sleep({duration})", **kwargs)
+                    for _ in range(times)
+                ]
+            )
+        else:
+            for _ in range(times):
+                await self.asql(f"select pg_sleep({duration})", **kwargs)
+
+    def test(self, **kwargs):
+        """Test if you can connect"""
+        return self.sql("select 1", **kwargs)
+
+    def atest(self, **kwargs):
+        """Test if you can connect asynchronously"""
+        return self.asql("select 1", **kwargs)
+
+    def psql_test(self, **kwargs):
+        """Test if you can connect with psql instead of psycopg"""
+        return self.psql("select 1", **kwargs)
+
+    @contextmanager
+    def enable_firewall(self):
```

Comment:
```suggestion
    def _enable_firewall(self):
```

Prefix '_' indicates it is internal.